contrib/screener-api/client/client.go

```diff
@@ -54,6 +66,59 @@ func (c clientImpl) ScreenAddress(ctx context.Context, ruleset, address string)
 	return blockedRes.Blocked, nil
 }
 
+// BlackListBody is the json payload that represents a blacklisted address.
+type BlackListBody struct {
+	TypeReq string `json:"typereq"`
+	ID      string `json:"id"`
+	Data    string `json:"data"`
+	Address string `json:"address"`
+	Network string `json:"network"`
+	Tag     string `json:"tag"`
+	Remark  string `json:"remark"`
+}
+
+type blacklistResponse struct {
+	Status string `json:"status"`
+	Error  string `json:"error"`
+}
+
+func (c clientImpl) BlacklistAddress(ctx context.Context, body BlackListBody) (string, error) {
+	var blacklistRes blacklistResponse
+
+	// TODO: remove, just for testing purposes
+	// future, take it from some .env or something
+	appsecret := "appsecret"
```

Comment:
Here's how I'd approach this. The screener-api client stub is meant for testing and for other go packages to import (see [imports](https://pkg.go.dev/github.com/synapsecns/sanguine/contrib/screener-api/client?tab=importedby))

[Because of this, we should use an [option pattern](https://dev.to/kittipat1413/understanding-the-options-pattern-in-go-390c) to add auth as an optional param, the way we do with the [ethergo client](https://pkg.go.dev/github.com/synapsecns/sanguine/ethergo/client#DialBackend)

This auth can be applied to unauthenticated endpoints without breaking anything. 


The actual process of reading in these keys from the enviornment can happen in the app itself rather than the client stub.